asim: add mma skew experiment

tbg opened the pull request:
_Very_ draft.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152913)
<!-- Reviewable:end -->

tbg committed:
asim: allow {min_max}_key={min,max} shorthand

When running tests with constraints, it's nice to have the existing ranges match
_exactly_ the generated ranges. This typically requires covering the entire
keyspace, which previously required use of the secret constant "9999999" (I
might be off by a few 9s here but you get the idea).

With this commit, you can also specify `min` or `max` directly.

Epic: CRDB-49117

tbg committed:
wip

tbg committed:
mmaprototype: gracefully handle zero mean load

This can happen in the asim tests. It could conceivably also happen in
production.

tbg committed:
asim: fill gaps in and harmonize store/node log tags

Noticed that a bunch of them were missing, which made it hard to figure out who
was logging what.

sumeerbhola committed:
wip2

wenyihu6 committed:
unrelated config fix

wenyihu6 committed:
populate voter count as replica count if 0

wenyihu6 committed:
revert the span conig on constraint

wenyihu6 committed:
stores on the same node should not be excluded

wenyihu6 committed:
change constraint max key to be 5000,9999999999 since gen_ranges cover to max_key=max

tbg commented:
note to self: we have an existing test that has skewed load. See about making that one a bit more interesting instead of piling on another one. I think the test is `skewed_cpu_skewed_write`.
